### PR TITLE
Softening the density of states featurizers

### PR DESCRIPTION
**Pull request by dyllamt**
## Summary

* An exponential decay function was introduced to sample the band edge density of states smoothly.
* This affects both `DOSFeaturizer` and `Hybridization` (formerly `BandEdge`). The name change is consistent with the output of the featurizer.
* The number of significant features in `DOSFeaturizer` was changed to an entropy score to reflect the degree of hybridization at the band edge.
* Updated tests. The orbital scores before and after implementing the exponential decay function were similar, but the precise digits changed.

### CODE REVIEW

**Comment by dyllamt** [comment]
Linking to issue #246

**Comment by dyllamt** [comment]
Failing style test is for redefining the `featurize` method in `Hybridization` (formerly `BandEdge`).

I don't think that it affects the use of the featurizer.

**Comment by albalu** [comment]
Hi @dyllamt , yes I had that Codacy issue before too, you can ignore that one. Thank you for implementing this. Regarding the name Hybridization, it makes it clear that we can get hybridization but doesn't cover features such as cbm_p or Pb_d as opposed to cbm_sp (hybridization) that was why I thought about BandEdge or maybe BandEdgeCharacter; regardless, since it is under dos.py it implies that it looks at dos information.

**Comment by dyllamt** [comment]
Hmmm that's a good point that there is more than just hybridization features. Maybe we should break that up into two featurizers?

Then there would be:
1. DosFeaturizer - returns site and orbital projected DOS
2. DosOrbitals - returns orbital projected DOS and hybridizations
3. DosSpecie - returns s/p/d/f DOS for a particular species

**Comment by albalu** [comment]
I think "Hybridization" is better than having two separate classes.

File: matminer/featurizers/dos.py [hunk 1]
```diff
@@ -59,25 +56,30 @@ def featurize(self, dos):
...
                                        self.gaussian_smear)
 
         feat = OrderedDict()
         for ex in ['cbm', 'vbm']:
-            orbscores.sort(key=lambda x: x['{}_score'.format(ex)], reverse=True)
+            orbscores.sort(key=lambda x: x['{}_score'.format(ex)],
+                           reverse=True)
             scores = np.array([s['{}_score'.format(ex)] for s in orbscores])
-            feat['{}_nsignificant'.format(ex)] = len(scores[scores > self.significance_threshold])
+            print(scores)
```
Comment (albalu) [inline matminer/featurizers/dos.py]: debug print leftover

**Comment by dyllamt** [comment]
Good catch, removed the print statement from the pull request!

**Comment by albalu** [comment]
@computron , I don't have permission to merge this. Could you please if everything looks good?

**Comment by computron** [comment]
@albalu I made some permission changes, can you see if you can merge now?

**Comment by albalu** [comment]
Yes thank you both!

**Pull request merged**